src/components/ModalManager.js

```diff
@@ -26,10 +26,16 @@ const ModalManager = (props) => {
 
   const selectedModal = () => {
     switch (type) {
-      case "add":
+      case 'add':
         return <AddRoom />;
-      case "enter":
-        return <EnterRoom title={content.title} id={content.id} />;
+      case 'enter':
+        return (
+          <EnterRoom
+            title={content.title}
+            id={content.id}
+            protect={content.private}
```

Comment:
Protect only if user is not register and content is private:
`protect={unregistered && content.private}`